common/common.go

```diff
@@ -610,22 +610,40 @@ func GetDefaultDataDir(env string) string {
 	return path.Join(os.ExpandEnv("$HOME"), ".gocryptotrader")
 }
 
-// CheckDir checks to see if a particular directory exists
-// and attempts to create it if desired, if it doesn't exist
-func CheckDir(dir string, create bool) error {
+// CreateDir creates a directory based on the supplied parameter
+func CreateDir(dir string) error {
 	_, err := os.Stat(dir)
 	if !os.IsNotExist(err) {
 		return nil
 	}
 
-	if !create {
-		return fmt.Errorf("directory %s does not exist. Err: %s", dir, err)
-	}
-
 	log.Warnf("Directory %s does not exist.. creating.", dir)
-	err = os.MkdirAll(dir, 0777)
+	return os.MkdirAll(dir, 0770)
+}
+
+// ListAllDir lists all the directories and files in an array
+func ListAllDir(directory string) ([]string, error) {
+	var list []string
+	err := filepath.Walk(directory, func(path string, info os.FileInfo, err error) error {
+		if !info.IsDir() {
+			list = append(list, path)
+		}
+		return nil
+	})
+	return list, err
+}
+
+// ChangePerm changes file permissions in a given directory to 0770
+func ChangePerm(env string) error {
+	a, err := ListAllDir(GetDefaultDataDir(env))
```

Comment:
To make this a little more generic we could remove the usage of GetDefaultDataDir here that way it supports any folder and calls to it could pass the path which would also mean remaining the param from env to something more generic